### PR TITLE
[Merged by Bors] - improve error for packet ignore outgoing

### PR DESCRIPTION
**Pull request by aviramha**
agent: Return better error message when failing to use `PACKET_IGNORE_OUTGOING` flag.

Part of #912

### CODE REVIEW

File: mirrord/agent/src/error.rs [hunk 1]
```diff
@@ -105,6 +105,9 @@ pub enum AgentError {
 
     #[error("An internal invariant of the agent was violated, this should not happen.")]
     AgentInvariantViolated,
+
+    #[error("Failed to set socket flag PACKET_IGNORE_OUTGOING, this might be due to kernel version before 4.20")]
```
Comment (meowjesty) [inline mirrord/agent/src/error.rs:109]: Will this display the "raw" error also?

Comment (aviramha) [reply mirrord/agent/src/error.rs:109]: No, good catch.

**Review by meowjesty — APPROVED** [review]
🐻 👍

**Pull request closed without merging**